### PR TITLE
Initial swerve drive implementation

### LINKED ISSUE #42: Prototype a drivetrain ROS node
Here are some resources to help with swerve drive kinematics:
-https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/kinematics/SwerveDriveKinematics.html
-https://github.com/ros-controls/kinematics_interface

### PR DESCRIPTION
**Pull request by Isopod00**
The turning of the modules still needs to be optimized and methods for the absolute encoders need to be written, but other than that this is basically complete and we've tested it.

### CODE REVIEW

File: src/drivetrain/drivetrain/drivetrain_node.py [hunk 1]
```diff
@@ -30,41 +62,64 @@ def __init__(self):
...
-        )
+    def drive(self, forward_power: float, horizontal_power: float, turning_power: float) -> None:
+        """This method drives the robot with the desired forward, horizontal and turning power."""
+
+        # Essentially, we need to take the forward_power (y), horizontal_power(x), and turning_power (z)
+        # and compute the what the angles and powers of all 4 swerve modules should be.
+
+        # Vector layouts = [Drive Power, Drive Direction(Degrees from forwards going counterclockwise)]
+
+        wheel_base = 1  # Half of the wheelbase length # TODO: change this value when we know the final measurements
```
Comment (jmblixt3) [inline src/drivetrain/drivetrain/drivetrain_node.py]: Move these to a config file

File: src/drivetrain/drivetrain/drivetrain_node.py [hunk 2]
```diff
@@ -30,41 +62,64 @@ def __init__(self):
         self.srv_stop = self.create_service(Stop, "drivetrain/stop", self.stop_callback)
         self.srv_drive = self.create_service(Drive, "drivetrain/drive", self.drive_callback)
 
-        # Define motor CAN IDs here
-        self.FRONT_LEFT_DRIVE = 1
-        self.BACK_LEFT_DRIVE = 4
-        self.FRONT_RIGHT_DRIVE = 3
-        self.BACK_RIGHT_DRIVE = 2
+        # Assigning can_id motor to each swerve module
+        self.back_left_drive = 1
```
Comment (jmblixt3) [inline src/drivetrain/drivetrain/drivetrain_node.py]: Move these to config file

**Review by jmblixt3 — CHANGES_REQUESTED** [review]
Move some things to config file

File: src/drivetrain/drivetrain/drivetrain_node.py (hunk 1 above)
Comment (Isopod00) [reply src/drivetrain/drivetrain/drivetrain_node.py]: done

File: src/drivetrain/drivetrain/drivetrain_node.py (hunk 2 above)
Comment (Isopod00) [reply src/drivetrain/drivetrain/drivetrain_node.py]: done

File: src/drivetrain/drivetrain/drivetrain_node.py [hunk 3]
```diff
@@ -62,41 +62,61 @@ def __init__(self):
...
+        self.FRONT_LEFT_TURN = self.get_parameter("FRONT_LEFT_TURN").value
+        self.BACK_RIGHT_DRIVE = self.get_parameter("BACK_RIGHT_DRIVE").value
+        self.BACK_RIGHT_TURN = self.get_parameter("BACK_RIGHT_TURN").value
+        self.FRONT_RIGHT_DRIVE = self.get_parameter("FRONT_RIGHT_DRIVE").value
+        self.FRONT_RIGHT_TURN = self.get_parameter("FRONT_RIGHT_TURN").value
+        self.WHEEL_BASE = self.get_parameter("WHEEL_BASE").value
+        self.TRACK_WIDTH = self.get_parameter("TRACK_WIDTH").value
+
+        # Print the ROS Parameters to the terminal below #
+        print("BACK_LEFT_DRIVE has been set to:", self.BACK_LEFT_DRIVE)
```
Comment (jmblixt3) [inline src/drivetrain/drivetrain/drivetrain_node.py:90]: Can you not use print statements, they won't be visible in a launch file

**Review by jmblixt3 — CHANGES_REQUESTED** [review]
Do that then ill rebase and merge

File: src/drivetrain/drivetrain/drivetrain_node.py (hunk 3 above)
Comment (jmblixt3) [reply src/drivetrain/drivetrain/drivetrain_node.py:90]: Moved to #98

**Review by jmblixt3 — APPROVED** [review]

**Pull request merged**